packages/beacon-transport-walletconnect/src/communication-client/WalletConnectCommunicationClient.ts

```diff
@@ -608,6 +584,23 @@ export class WalletConnectCommunicationClient extends CommunicationClient {
     // }
   }
 
+  private async sendResponse(session: SessionTypes.Struct, partialResponse: BeaconResponseInputMessage) {
+    const response: BeaconResponseMessage = {
+      ...partialResponse,
+      version: '2',
+      senderId: session.peer.publicKey
+    } as BeaconResponseMessage
```

Comment:
This change means that until now, the response that was serialised was always wrong / incomplete? Because it never contained the version or senderId?

Reply:
Yes, this is exactly what I found out. The `version` wasn't that important as there's a distinguished check [only for v3 messages](https://github.com/airgap-it/beacon-sdk/blob/develop/packages/beacon-dapp/src/dapp-client/DAppClient.ts#L227) and if the condition isn't met, [everything just falls into the v2 case](https://github.com/airgap-it/beacon-sdk/blob/develop/packages/beacon-dapp/src/dapp-client/DAppClient.ts#L286). The problem, however, was indeed the `senderId` which was missing in, for example, [the permission response](https://github.com/airgap-it/beacon-sdk/blob/develop/packages/beacon-dapp/src/dapp-client/DAppClient.ts#L950) preventing the SDK from [finding correct app metadata](https://github.com/airgap-it/beacon-sdk/blob/develop/packages/beacon-dapp/src/dapp-client/DAppClient.ts#L1415).